keras_cv/layers/preprocessing/random_translation.py

```diff
@@ -0,0 +1,256 @@
+        bounding_box_format=None,
+        **kwargs
+    ):
+        super().__init__(seed=seed, **kwargs)
+        self.seed = seed
+        if x_factor is not None:
+            self.x_factor = preprocessing.parse_factor(
+                x_factor,
+                min_value=-1.0,
+                max_value=1.0,
+                param_name="x_factor",
+                seed=seed,
+                range_mode="symmetric",
+            )
+        else:
+            self.x_factor = x_factor
+        if y_factor is not None:
+            self.y_factor = preprocessing.parse_factor(
+                y_factor,
+                min_value=-1.0,
+                max_value=1.0,
+                param_name="y_factor",
+                seed=seed,
+                range_mode="symmetric",
+            )
+        else:
+            self.y_factor = y_factor
+        if x_factor is None and y_factor is None:
+            warnings.warn(
+                "RandomTranslation received both `x_factor=None` and `y_factor=None`.  "
+                "As a result, the layer will perform no augmentation."
+            )
+        self.interpolation = interpolation
+        self.fill_mode = fill_mode
+        self.fill_value = fill_value
+        self.seed = seed
+        self.bounding_box_format = bounding_box_format
+
+    def augment_label(self, label, transformation, **kwargs):
+        return label
+
+    def augment_image(self, image, transformation, **kwargs):
+        return self._translate_image(image, transformation)
+
+    def get_random_transformation(self, **kwargs):
+        translate_horizontal = 0.0
+        translate_vertical = 0.0
+        if self.x_factor:
+            translate_horizontal = self.x_factor()
+        if self.y_factor:
+            translate_vertical = self.y_factor()
+        return {
+            "translate_horizontal": tf.cast(translate_horizontal, dtype=tf.float32),
+            "translate_vertical": tf.cast(translate_vertical, dtype=tf.float32),
+        }
+
+    def _translate_image(self, image, transformation):
+        image = preprocessing.ensure_tensor(image, self.compute_dtype)
+        image_shape = image.shape
+        image_width = tf.cast(image_shape[1], tf.float32)
+        image_height = tf.cast(image_shape[0], tf.float32)
+        element = tf.stack(
+            [
+                transformation["translate_horizontal"] * image_width,
+                transformation["translate_vertical"] * image_height,
+            ],
+        )
+        output = preprocessing.transform(
+            tf.expand_dims(image, 0),
+            preprocessing.get_translation_matrix(tf.expand_dims(element, axis=0)),
+            fill_mode=self.fill_mode,
+            fill_value=self.fill_value,
+            interpolation=self.interpolation,
+        )
+        output = tf.squeeze(output, 0)
+        output.set_shape(image_shape)
+        return output
+
+    @staticmethod
+    def _translate_bounding_boxes_horizontal(bounding_boxes, dx):
```

Comment:
I think this code may be easier to follow if we perform horizontal and vertical translation together in one method, e.g:

```
 def _translate_bounding_boxes_vertical(bounding_boxes, dy):
        x0, y0, x1, y1, rest = tf.split(
            bounding_boxes, [1, 1, 1, 1, bounding_boxes.shape[-1] - 4], axis=-1
        )
        output = tf.stack(
            [
                x0 + dx,
                y0 + dy,
                x1 + dx,
                y1 + dy,
                rest,
            ],
            axis=-1,
        )
        output = tf.squeeze(output, axis=1)
        return output
```